controller/modules/ph/probe.go

```diff
@@ -181,6 +187,34 @@ func (c *Controller) Calibrate(id string, ms []hal.Measurement) error {
 	return c.c.Store().Update(CalibrationBucket, p.ID, ms)
 }
 
+func (c *Controller) CalibratePoint(id string, point CalibrationPoint) error {
+	if point.Expected > 14 || point.Expected <= 0 {
+		return fmt.Errorf("Invalid expected calibration value %f. Valid values are above 0  and below 14", point.Expected)
+	}
+
+	p, err := c.Get(id)
+	if err != nil {
+		return err
+	}
+	if p.Enable {
+		return fmt.Errorf("Probe must be disabled from automatic polling before running calibration")
+	}
+
+	var calibration []hal.Measurement
+
+	//Append to existing calibration unless the point is the mid point.
+	//Receiving a mid point calibration resets the calibration process.
+	if point.Type != "mid" {
```

Comment:
do we need to chec the point.Type? we can just infer it from introspecting the number of calibration points? or is it purely for the step wise wizard like UX ?

Reply:
I'm open to other opinions on this, but here are my thoughts.

I think we need point.type for a couple of reasons.
1. If you only calibrate the mid point, then the calibration factory will use single point calibration.  Without point.type, I won't know to discard any existing calibration measurements.
2. If we support 3 point calibration in the future, I think it would be more difficult to make inferences based on existing calibration points. 

I could alternatively store the calibration measurements on the UI, then ask the user to commit them once they are happy.   That's not my preference since it changes the current expectations and could possibly result in a confusing user experience if the don't submit the measurements after collecting the points.